README.md

```diff
@@ -1 +1,244 @@
+within the Sdk.
+
+An example implementation of this would be as follows 
+
+```rust
+use alvarium_annotator::{
+    Annotation, Annotator, constants, derive_hash, serialise_and_sign,
+    constants::AnnotationType,
+};
+use alvarium_rust_sdk::{
+    config::{self, Signable}, 
+    factories::{new_hash_provider, new_signature_provider}, 
+    providers::sign_provider::SignatureProviderWrap
+};
+
+/// Defines a new annotator type that will implement the Annotator trait
+pub struct ThresholdAnnotator {
+    /// Hashing algorithm used for checksums
+    hash: constants::HashType,
+    /// Type of annotation (a wrapper around a string definition)
+    kind: AnnotationType,
+    /// Signature provider for signing data
+    sign: SignatureProviderWrap,
+    /// Threshold limits for custom annotation
+    range: Range<u8>,
+}
+
+impl ThresholdAnnotator {
+    pub fn new(cfg: &config::SdkInfo, range: Range<u8>) -> Result<impl Annotator<Error = alvarium_rust_sdk::errors::Error>> {
+        Ok(ThresholdAnnotator {
+            hash: cfg.hash.hash_type.clone(),
+            kind: AnnotationType("threshold".to_string()),
+            sign: new_signature_provider(&cfg.signature)?,
+            range,
+        })
+    }
+
+}
+
+/// Implementation of the annotate() function for generating a threshold Annotation
+impl Annotator for ThresholdAnnotator {
+    type Error = alvarium_rust_sdk::errors::Error;
+    fn annotate(&mut self, data: `&[u8]`) -> alvarium_rust_sdk::errors::Result<Annotation> {
+        let hasher = new_hash_provider(&self.hash)?;
+        let signable: Signable = serde_json::from_slice(data)?;
+        let key = derive_hash(hasher, signable.seed.as_bytes());
+        
+        match gethostname::gethostname().to_str() {
+            Some(host) => {
+                let reading: std::result::Result<SensorReading, serde_json::Error> = serde_json::from_slice(data);
+                let within_threshold = match reading {
+                    Ok(reading) => {
+                        let reading: SensorReading = serde_json::from_str(&signable.seed).unwrap();
+                        reading.value <= self.range.end && reading.value >= self.range.start
+                    },
+                    Err(_) => false
+                };
+
+                let mut annotation = Annotation::new(&key, self.hash.clone(), host, self.kind.clone(), within_threshold);
+                let signature = serialise_and_sign(&self.sign, &annotation)?;
+                annotation.with_signature(&signature);
+                Ok(annotation)
+            },
+            None => {
+                Err(alvarium_rust_sdk::errors::Error::NoHostName.into())
+            }
+        }
+    }
+}
+```
+
+#### Custom Stream Providers
+The base SDK includes an Mqtt and Demia Streams provider, but new stream providers can be created 
+using the alvarium-annotator [Publisher](https://github.com/DyrellC/alvarium-annotator/blob/main/src/providers.rs#L124)
```

Comment:
@DyrellC I guess just do a scan for all DyrellC github references and replace them